### PR TITLE
Fix install repo firing twice (#3678)

### LINKED ISSUE #3678: UI: app repository Install Repo button fires twice (sometimes)
### Description:

I ran several times into the use case where clicking the "Install Repo" button to add a new app repository is being executed twice, resulting in an error message being displayed.
The first execution seems to properly create the app repository, and does not seem to be affected by the second execution.

### Describe the results you expected:

Any submit button should be guarded from firing multiple times.

### PR DESCRIPTION
**Pull request by castelblanque**
Signed-off-by: Rafa Castelblanque <rcastelblanq@vmware.com>

### Description of the change

Fixes the problem of "Install Repo" button being fired multiple times.

### Benefits

Repo installation process is not triggered more than once.

### Possible drawbacks

N/A

### Applicable issues

- fixes #3678

### Additional information

Makes use of [React's useRef](https://reactjs.org/docs/hooks-reference.html#useref)

### CODE REVIEW

File: dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx [hunk 1]
```diff
@@ -49,6 +49,7 @@ const TYPE_OCI = "oci";
 
 export function AppRepoForm(props: IAppRepoFormProps) {
   const { onSubmit, onAfterInstall, namespace, kubeappsNamespace, repo, secret } = props;
+  const workingRef = useRef(false);
```
Comment (antgamdia) [inline dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx:52]: I'd maybe choose another name, a more self-explanatory one like  `isInstallingRef` ?

**Review by antgamdia — APPROVED** [review]
Thanks for fixing that issue! Great work :)

File: dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx [hunk 2]
```diff
@@ -620,7 +627,7 @@ export function AppRepoForm(props: IAppRepoFormProps) {
         </Alert>
       )}
       <div className="margin-t-xl">
-        <CdsButton disabled={validating} onClick={install}>
+        <CdsButton type="submit" disabled={validating}>
```
[the remark this answers is not in the document]
Comment (castelblanque) [reply dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx:630]: U nailed it! `install` function was being called twice: here and inside form's onSubmit logic.

File: dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx (hunk 1 above)
Comment (castelblanque) [reply dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx:52]: You are right, will rename it. Initially I wanted to make the solution something general for all forms+buttons :)

**Pull request merged**